### PR TITLE
Fixing #11 (enum name harmony) and #12 (pass down transaction)

### PR DESCRIPTION
**Pull request by cyrilchapon**
Fixing #11
Fixing #12

Transaction passing down require additional manual testing

### CODE REVIEW

**Comment by TheOptimisticFactory** [comment]
I've taken the liberty of publishing your changeset under `"@joethefkingfrypan/sequelize-replace-enum-postgres": "^1.6.0-rc1"` until your PR gets merged :smiley:

**Comment by cyrilchapon** [comment]
Great news dude :)

**Comment by cyrilchapon** [comment]
(I'll close issues when merged into release ?)

**Comment by cyrilchapon** [comment]
NB: One thing to note is that I'm using a not-very-documented sequelize Feature : when you pass a `sequelizeOptions` to `sequelize.transaction(sequelizeOptions` (before the callback); and that `sequelizeOptions` contain a transaction; then the created transaction is kind of a *"sub-transaction"* (in postgres, it actually creates a savepoint).

NB2: I'm not `.catch`ing anything on the promise neither on the sub-transaction, and letting the error bubble up intentionnaly. This will defer the handling of the error to the calling context. For example :

## intentionally fail the parent transaction also

```
await sequelizeTransaction(async parentTransaction => {
  await replaceEnum({ transaction: parentTransaction })
})
```

## intentionally handle the error, and resolve parent transaction

```
await sequelizeTransaction(async parentTransaction => {
  try {
    await replaceEnum({ transaction: parentTransaction })
  } catch (err) {
    console.error('non important error')

    // Here, catching will cause
    // function to resolve
    // and parentTransaction to commit
    // but keeping the replaceEnum "sub-transaction"
    // to rollback
  }
})
```

**Comment by abelosorio** [comment]
Hi @cyrilchapon @TheOptimisticFactory, have you guys tested this PR? Currently, I don't have time for this project, so I'm relying on you for this. Let me know if this is fully tested and I'll merge it.

**Comment by cyrilchapon** [comment]
Hey @abelosorio

I wrote a couple of unit tests, but that's the kind of features I usually test with "side effects tests" (against a real database instead of a mocked one).

**Comment by cyrilchapon** [comment]
To be perfectly honest I would not be 100% confident deploying it on production (though, I did). I think this is so specific that it needs to be unit tested agasint a real database

File: src/index.js [hunk 1]
```diff
@@ -20,46 +20,62 @@ export default (args) => {
...
     enumName = `enum_${tableName}_${columnName}`
   } = args;
 
   const newEnumName = `${enumName}_new`;
 
-  return queryInterface.sequelize.transaction((t) => {
-    const sequelizeOptions = { transaction: t };
+  return queryInterface.sequelize.transaction(sequelizeOptions, (t) => {
+    const newSequelizeOptions = {
+      ...(sequelizeOptions || {}),
```
Comment (abelosorio) [inline src/index.js]: `sequelizeOptions` default to `{}`, so why is this needed here?

**Comment by cyrilchapon** [comment]
To override explicitly null / 0 / "" passed values, but this is overkill indeed ☺️

**Comment by abelosorio** [comment]
> To override explicitly null / 0 / "" passed values, but this is overkill indeed

Yeah, I think so as well. Can you remove it so I can merge this?

Thanks

**Review by abelosorio — APPROVED** [review]

**Comment by abelosorio** [comment]
Hi @cyrilchapon! I haven't had the time to look at this in detail.
